packages/core-js/modules/es.string.match.js

```diff
@@ -1,11 +1,42 @@
 'use strict';
+
+var anObject = require('../internals/an-object');
+var toLength = require('../internals/to-length');
+var advanceStringIndex = require('../internals/advance-string-index');
+var nativeExec = RegExp.prototype.exec;
+
 // @@match logic
 require('../internals/fix-regexp-well-known-symbol-logic')('match', 1, function (defined, MATCH, nativeMatch) {
-  // `String.prototype.match` method
-  // https://tc39.github.io/ecma262/#sec-string.prototype.match
-  return [function match(regexp) {
-    var O = defined(this);
-    var matcher = regexp == undefined ? undefined : regexp[MATCH];
-    return matcher !== undefined ? matcher.call(regexp, O) : new RegExp(regexp)[MATCH](String(O));
-  }, nativeMatch];
+  return [
+    // `String.prototype.match` method
+    // https://tc39.github.io/ecma262/#sec-string.prototype.match
+    function match(regexp) {
+      var O = defined(this);
+      var matcher = regexp == undefined ? undefined : regexp[MATCH];
+      return matcher !== undefined ? matcher.call(regexp, O) : new RegExp(regexp)[MATCH](String(O));
+    },
+    // `RegExp.prototype[@@match]` method
+    // https://tc39.github.io/ecma262/#sec-regexp.prototype-@@match
+    function Symbol$match(regexp) {
+      if (regexp.exec === nativeExec) return nativeMatch.call(this, regexp);
+
+      var rx = anObject(regexp);
+      var S = String(this);
+
+      if (!rx.global) return rx.exec(S);
+
+      var fullUnicode = rx.unicode;
+      rx.lastIndex = 0;
+      var A = [];
+      var n = 0;
+      var result;
+      while ((result = rx.exec(S)) !== null) {
```

Comment:
In [`RegExpExec`](https://tc39.github.io/ecma262/#sec-regexpexec), own `.exec` should be called only if it's callable. Mabe makes sense to implement `RegExpExec` abstract operation?